server/test/auth.spec.js

```diff
@@ -0,0 +1,89 @@
+const chai = require("chai");
+const chaiHttp = require("chai-http");
+const { server } = require("../app.js");
+const dotenv = require('dotenv')
+const expect = require("chai").expect;
+
+dotenv.config();;
+
+chai.should();
+chai.use(chaiHttp);
+
+const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjYxMjdmYjA3OTNhYTZmM2ExYzkyM2ZiZiIsImlhdCI6MTYzMDAxMDExOSwiZXhwIjoxNjMwNjE0OTE5fQ.EE8omcwEkQkCEJUEMyY2ODQgBHOZk3g5spRPwQ6jynM';
+
+
+// Test for login
+describe("/POST login", () => {
+  it("it should  return 200 if credentials are valid or 401 if not", (done) => {
+    chai
+      .request(server)
+      .post(`/auth/login`)
+      .send({ email: "jondoe@vmail.com", password: "12345678" })
+      .end((err, res) => {
+        expect(res.status).to.be.oneOf([200, 401]);
+        done();
+      });
+  });
+});
+
+// Test for logout route
+describe("/GET logout", () => {
+  it("it should return 205 and message", (done) => {
+    chai
+      .request(server)
+      .get(`/auth/logout`)
+      .end((err, res) => {
+        res.should.have.status(205);
+        res.body.should.have
+          .property("message")
+          .eql("You have successfully logged out");
+        done();
+      });
+  });
+});
+
+// Test for register
+describe("/POST register", () => {
+  it("it should return 201 or 400 is email or username already exists", (done) => {
+    chai
+      .request(server)
+      .post(`/auth/register`)
+      .send({ email: "jondoe@vmail.com", password: "12345678", username: "joDe" })
+      .end((err, res) => {
+        expect(res.status).to.be.oneOf([201, 400]);
+        done();
```

Comment:
Same here, this test case really doesn't make sense to be combined, I would separate.

Reply:
Same as above case, this test cannot have same result more than once, you can only register once so.
Please see https://hatchways.slack.com/archives/C02BL4UC3SA/p1630081896021300 this is the problem I'm trying to solve.